components/layout_2020/display_list/stacking_context.rs

```diff
@@ -420,38 +509,32 @@ impl StackingContext {
         // Let’s find the corresponding fragment.
 
         // The fragment generated by the root element is the first one here, unless…
-        let first_if_any = self.fragments.first().or_else(|| {
+        let first_if_any = self.contents.first().or_else(|| {
             // There wasn’t any `StackingContextFragment` in the root `StackingContext`,
             // because the root element generates a stacking context. Let’s find that one.
-            self.stacking_contexts
+            self.real_stacking_contexts_and_positioned_stacking_containers
                 .first()
                 .and_then(|first_child_stacking_context| {
-                    first_child_stacking_context.fragments.first()
+                    first_child_stacking_context.contents.first()
                 })
         });
 
-        macro_rules! debug_panic {
-            ($msg: expr) => {
-                if cfg!(debug_assertions) {
-                    panic!($msg)
-                }
-            };
-        }
-
         let first_stacking_context_fragment = if let Some(first) = first_if_any {
             first
         } else {
             // This should only happen if the root element has `display: none`
-            debug_panic!("`CanvasBackground::for_root_element` should have returned `style: None`");
-            return;
+            panic!("`CanvasBackground::for_root_element` should have returned `style: None`");
```

Comment:
I think this should fail gracefully instead of panic. I believe we need to start cleaning up as many panics as possible.

Reply:
done in 43692821f9bccde5eda1b5651e21c535bfaebe9d